tssc/step_implementer.py

```diff
@@ -90,124 +158,172 @@ def step_name(cls):
-        kwargs : dict
-            Arbitrary arguments passed in via runtime to merge with the step_config
-            to use when running the step.
+        step_config_runtime_overrides : dict, optional
+            Configuration for the step passed in at runtime when the step was invoked that will
+            override step configuration coming from any other source.
         """
 
-        StepImplementer.__print_section_title("TSSC Step Start - {}".format(self.step_name()))
-        runtime_step_config = {**global_step_config_defaults, **self.step_config, **kwargs}
-        self._validate_runtime_step_config(runtime_step_config)
+        step_config_runtime_overrides = {} if step_config_runtime_overrides is None \
+                                            else step_config_runtime_overrides
 
-        StepImplementer.__print_data('Global Default Static Step Configuration', \
-          global_step_config_defaults)
-        StepImplementer.__print_data('Static Step Configuration', self.step_config)
-        StepImplementer.__print_data('Dynamic Step Configuration', kwargs)
-        StepImplementer.__print_data('Runtime Step Configuration', runtime_step_config)
+        StepImplementer.__print_section_title("TSSC Step Start - {}".format(self.step_name()))
 
+        # print information about the static step configuration
+        StepImplementer.__print_data(
+            "Step Implementer Configuration Defaults",
+            self.step_implementer_config_defaults())
+        StepImplementer.__print_data(
+            "Global Configuration Defaults",
+            self.global_config_defaults)
+        StepImplementer.__print_data(
+            "Global Environment Configuration Defaults",
+            self.global_environment_config_defaults)
+        StepImplementer.__print_data(
+            "Step Configuration",
+            self.step_config)
+        StepImplementer.__print_data(
+            "Step Environment Configuration",
+            self.step_environment_config)
+        StepImplementer.__print_data(
+            "Step Environment Configuration",
+            self.step_environment_config)
+        StepImplementer.__print_data(
+            "Step Configuration Runtime Overrides",
+            step_config_runtime_overrides)
+
+        # create the munged runtime step configuration and print
+        runtime_step_config = self.__create_runtime_step_config(step_config_runtime_overrides)
+        StepImplementer.__print_data(
+            "Runtime Step Configuration",
+            runtime_step_config)
+
+        # validate the runtime step configuration, run the step, and save the results
+        self._validate_runtime_step_config(runtime_step_config)
         results = self._run_step(runtime_step_config)
         self.write_results(results)
 
+        # print the step run results
         StepImplementer.__print_section_title("TSSC Step Results - {}".format(self.step_name()))
```

Comment:
I'd reformat this to look like the above for consistency.

Reply:
ummm. i dont understand. im just resolving this.